myia/compile/utils.py

```diff
@@ -1,4 +1,18 @@
 """Utility functions for graph compilation and code generation."""
+from dataclasses import dataclass
+
+from ..abstract import AbstractValue
+from .backends import Backend
+
+
+@dataclass(frozen=True)
+class BackendValue:
+    """Class that represents a value in a backend."""
+
+    value: object
+    orig_t: AbstractValue
+    vm_t: AbstractValue
+    backend: Backend
```

Comment:
Since instances of this class will be returned to the user, its attributes should be obfuscated, and it should define `__getattr__` and other builtins to raise a helpful exception telling them to use `from_device`. `from_device` should probably be a method, too.

Reply:
I could definitely make `from_device` a method on this.

I'm less sure about overriding `__getattr__` and friends to warn people.

Reply:
People are likely to try to access `model.layers` or `model.W`, and `'BackendValue' object has no attribute 'W'` is hardly helpful.

Reply:
I've added `__getattr__` with a suitable error message.